LEAF_Nexus/scripts/refreshOrgchartEmployees.php

```diff
@@ -31,81 +31,114 @@
-
-	#used to disable if not found in national
+    $sql2 = "UPDATE employee
+            SET lastName=:lastName,
+            firstName=:firstName,
+            middleName=:midInit,
+            phoneticFirstName=:phoneticFname,
+            phoneticLastName=:phoneticLname,
+            domain=:domain,
+            deleted=:deleted,
+            lastUpdated=:lastUpdated
+            WHERE userName=:userName";
+
+    #used to disable if not found in national
     $sql3 = "UPDATE employee
-        SET deleted=:deleted
-        WHERE userName=:userName";
+            SET deleted=:deleted
+            WHERE userName=:userName";
 
-	$res = $phonedb->prepared_query($sql, $vars);
+    $res = $phonedb->prepared_query($sql, $vars);
 
-	if (count($res) == 0){
-	    //if there is no record in nat, disable the account.
-	    $vars = array(
-	        ':userName' => $userName,
+    if (count($res) == 0) {
+        //if there is no record in nat, disable the account.
+        $vars = array(
+            ':userName' => $userName,
             ':deleted' => time()
         );
-	    $db->prepared_query($sql3, $vars);
+        $db->prepared_query($sql3, $vars);
+    }
+
+    if (count($res) > 0) {
+        $vars = array(
+            ':userName' => $res[0]['userName'],
+            ':lastName' => $res[0]['lastName'],
+            ':firstName' => $res[0]['firstName'],
+            ':midInit' => $res[0]['middleName'],
+            ':phoneticFname' => $res[0]['phoneticFirstName'],
+            ':phoneticLname' => $res[0]['phoneticLastName'],
+            ':domain' => $res[0]['domain'],
+            ':deleted' => $res[0]['deleted'],
+            ':lastUpdated' => $res[0]['lastUpdated']
+        );
+
+        // sets local employee table
+        $db->prepared_query($sql2, $vars);
+
+        // sets local employee_data table
+        updateEmployeeData($res[0]['empUID'], $empUID);
     }
 
-	if (count($res) > 0) {
-		$vars = array(
-				':userName' => $res[0]['userName'],
-				':lastName' => $res[0]['lastName'],
-				':firstName' => $res[0]['firstName'],
-				':midInit' => $res[0]['middleName'],
-				':phoneticFname' => $res[0]['phoneticFirstName'],
-				':phoneticLname' => $res[0]['phoneticLastName'],
-				':domain' => $res[0]['domain'],
-				':deleted' => $res[0]['deleted'],
-				':lastUpdated' => $res[0]['lastUpdated']
-		);
-
-		// sets local employee table
-		$db->prepared_query($sql2, $vars);
-
-		// sets local employee_data table
-		updateEmployeeData($res[0]['empUID'], $empUID);
-	}
+
+    if (count($res) > 0) {
+        $currentTime = time();
+        $oneWeekAdded = $res[0]['lastUpdated'] + 604800;
+
+        if ($currentTime > $oneWeekAdded) {
+            // user is disabled if userUpdated is lower than the current time
+            $vars = array(
+                ':deleted' => $currentTime,
+                ':userName' => $userName
+            );
+            $sql = "UPDATE employee
+                    SET userDisabled=:deleted
```

Comment:
Employee table field name is 'deleted'.  Also, if we don't want to process accounts that have a 'lastUpdated' value of 0, we could check that and only go forward with the rest if it isn't 0, rather than processing it and then updating it again.  Not sure if such accounts might already be deleted for other reasons - if so, we would not want to re-enable them.

Reply:
Thank you for finding the issue. Please review it again, thank you!

Reply:
SET deleted=:deleted